drivers/sensor/bh1749/bh1749.h

```diff
@@ -0,0 +1,125 @@
+#define BH1749_MANUFACTURER_ID                          0x92
+
+/* BH1749_SYSTEM_CONTROL */
+#define BH1749_SYSTEM_CONTROL_PART_ID_Msk               GENMASK(5, 0)
+#define BH1749_SYSTEM_CONTROL_PART_ID                   0x0D
+#define BH1749_SYSTEM_CONTROL_SW_RESET_Msk              BIT(6)
+#define BH1749_SYSTEM_CONTROL_SW_RESET                  BIT(6)
+#define BH1749_SYSTEM_CONTROL_INT_RESET_Msk             BIT(7)
+#define BH1749_SYSTEM_CONTROL_INT_RESET                 BIT(7)
+
+/* BH1749_MODE_CONTROL1 */
+#define BH1749_MODE_CONTROL1_MEAS_MODE_Msk              GENMASK(2, 0)
+#define BH1749_MODE_CONTROL1_MEAS_MODE_120MS            0x02
+#define BH1749_MODE_CONTROL1_MEAS_MODE_240MS            0x03
+#define BH1749_MODE_CONTROL1_MEAS_MODE_35MS             0x05
+
+#define BH1749_MODE_CONTROL1_RGB_GAIN_Msk               GENMASK(4, 3)
+#define BH1749_MODE_CONTROL1_RGB_GAIN_1X                ((0x01) << 3)
+#define BH1749_MODE_CONTROL1_RGB_GAIN_32X               ((0x03) << 3)
+
+#define BH1749_MODE_CONTROL1_IR_GAIN_Msk                GENMASK(6, 5)
+#define BH1749_MODE_CONTROL1_IR_GAIN_1X                 ((0x01) << 5)
+#define BH1749_MODE_CONTROL1_IR_GAIN_32X                ((0x03) << 5)
+
+/* BH1749_MODE_CONTROL2 */
+#define BH1749_MODE_CONTROL2_RGB_EN_Msk                 BIT(4)
+#define BH1749_MODE_CONTROL2_RGB_EN_ENABLE              BIT(4)
+#define BH1749_MODE_CONTROL2_RGB_EN_DISABLE             0x00
+
+#define BH1749_MODE_CONTROL2_VALID_Msk                  BIT(7)
+
+/* BH1749_INTERRUPT */
+#define BH1749_INTERRUPT_ENABLE_Msk                     BIT(0)
+#define BH1749_INTERRUPT_ENABLE_DISABLE                 0x00
+#define BH1749_INTERRUPT_ENABLE_ENABLE                  BIT(0)
+
+#define BH1749_INTERRUPT_LATCH                          BIT(4)
+
+#define BH1749_INTERRUPT_INT_SOURCE_Msk                 GENMASK(3, 2)
+#define BH1749_INTERRUPT_INT_SOURCE_RED                 ((0x00) << 2)
+#define BH1749_INTERRUPT_INT_SOURCE_GREEN               ((0x01) << 2)
+#define BH1749_INTERRUPT_INT_SOURCE_BLUE                ((0x02) << 2)
+
+#define BH1749_INTERRUPT_INT_STATUS_Msk                 BIT(7)
+
+/* BH1749_PERSISTENCE */
+#define BH1749_PERSISTENCE_PERSISTENCE_Msk              GENMASK(1, 0)
+#define BH1749_PERSISTENCE_PERSISTENCE_ACTIVE_END       0x00
+#define BH1749_PERSISTENCE_PERSISTENCE_UPDATE_END       0x01
+#define BH1749_PERSISTENCE_PERSISTENCE_4_SAMPLES        0x02
+#define BH1749_PERSISTENCE_PERSISTENCE_8_SAMPLES        0x03
+
+/* BH1749 RGB/IR SAMPLE POSITIONS  */
+#define BH1749_RGB_BYTE_POS_RED                         0
+#define BH1749_RGB_BYTE_POS_GREEN                       1
+#define BH1749_RGB_BYTE_POS_BLUE                        2
+#define BH1749_RGB_BYTE_POS_IR                          4
+
+/* Manufacturer ID */
+#define BH1749_MANUFACTURER_ID_DEFAULT                  0xE0
+
+struct bh1749_data {
+	struct device *i2c;
+	struct device *gpio;
+	struct gpio_callback gpio_cb;
+	struct k_work work;
+	struct device *dev;
+	u16_t sample_rgb_ir[5];
```

Comment:
You should define a macro for this `5` and use it here and also in the code that fetches the samples.

Reply:
this is declared u16 since color fields are 16bit, but the fetch reads bytes. I don't want to do e.g
`#define RGBIR_DATA_BYTES 10`
`u16_t sample_rgb_ir[RGBIR_DATA_BYTES/2];`
for instance. Could you help me find a smoother approach?

Reply:
What about using the following:
```
#define BH1749_SAMPLE_POS_RED    0
#define BH1749_SAMPLE_POS_GREEN  1
#define BH1749_SAMPLE_POS_BLUE   2
#define BH1749_SAMPLE_POS_IR     4
#define BH1749_SAMPLES_TO_FETCH  5
```
and `BH1749_SAMPLES_TO_FETCH*sizeof(u16_t)` in the sample fetching call?
Or `BH1749_SAMPLE_COUNT` maybe?

Reply:
`BH1749_SAMPLES_TO_FETCH` sounds good, since we're always fetching all of them